libpoolprotocols/stratum/EthStratumClient.cpp

```diff
@@ -554,11 +574,31 @@ void EthStratumClient::processReponse(Json::Value& responseObject)
 		if (!responseObject.isMember("method")) {
 
 			_id = responseObject.get("id", 0).asUInt();
-			_isSuccess = !responseObject.isMember("error");
+			_isSuccess = !responseObject.isMember("error") || responseObject.get("error", Json::Value::null).isNull();
 
 			if (!_isSuccess) {
-				if (responseObject.isMember("error")) {
-					_errReason = responseObject.get("error", "Unknown error").asString();
+
+				if (!responseObject.get("error", Json::Value::null).isNull()) {
+
+					if (responseObject["error"].isConvertibleTo(Json::ValueType::stringValue)) {
```

Comment:
This block is very similar, if not identical, to the one above. Perhaps make it into a method.

Reply:
Yep you're right.
Sometimes laziness afflicts me too. :)
Wilco.